File: easy/palindrome.rb
1000.downto(1) do |x|
    broke = false
    (2..x/2+1).each do |n|
        if x % n == 0
            broke = true
            break
        end
    end
    if not broke and x.to_s.reverse == x.to_s
        puts x
        break
    end
end

File: easy/palindrome.go
package main

import (
    "fmt"
    "math"
)

func reverse(s string) string {
    b := make([]byte, len(s));
    var j int = len(s) - 1;
    for i := 0; i <= j; i++ {
        b[j-i] = s[i]
    }
    return string (b);
}

func main() {
    for i := int64(1000); i > 0; i-- {
        max := i / 2 + 1
        broke := false
        for j := int64(2); j < max; j++ {
            if math.Mod(float64(i), float64(j)) == 0 {
                broke = true
                break
            }
        }
        str := fmt.Sprintf("%d", i)
        if !broke && str == reverse(str) {
            fmt.Println(str)
            break
        }
    }
}
